crates/matrix-sdk-crypto/src/identities/room_identity_state.rs

```diff
@@ -161,21 +184,57 @@ impl<R: RoomIdentityProvider> RoomIdentityState<R> {
         user_identity: &UserIdentity,
     ) -> Option<IdentityStatusChange> {
         if let UserIdentity::Other(_) = &user_identity {
-            // If the user's state has changed
-            let new_state = state_of(user_identity);
-            let old_state = self.known_states.get(user_id);
-            if new_state != old_state {
-                Some(self.set_state(user_identity.user_id(), new_state))
-            } else {
-                // Nothing changed
-                None
-            }
+            self.update_user_state_to(user_id, self.room.state_of(user_identity))
         } else {
             // Ignore updates to our own identity
             None
         }
     }
 
+    fn update_user_state_to(
```

Comment:
I wouldn't object to a doc-comment on this. In particular: does it actually change internal state, or just return a "change" object?

Reply:
Fixed in 8d136e7149